### PR TITLE
final validation form

### PR DESCRIPTION
**Pull request by KNT2302**

### CODE REVIEW

**Review by phuc-create — CHANGES_REQUESTED** [review]
field email can't fulfill

File: src/component/checkbox/FormCheckbox.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+const FormCheckbox = ({ checkboxs }) => {
+  return (
+    <div name="checkboxGroup" className="checkboxGroup">
+      <div>Your favorite</div>
+      {checkboxs.map((checkbox, index) => (
+        <div key={index}>
+          <label>
+            <Field type="checkbox" name="checkboxGroup" value={`${index + 1}`} />
+            {checkbox.title}
```
Comment (phuc-create) [inline src/component/checkbox/FormCheckbox.js:12]: put name as a prop

**Review by phuc-create — CHANGES_REQUESTED** [review]

File: src/component/fieldtext/InputField.js [hunk 2]
```diff
@@ -1,14 +1,26 @@
...
           <label htmlFor={name}>{label}</label>
-          <input type={type} {...field} onChange={(e) => handleChangeInput(e, handleChange)} id={name}></input>
+          <input
+            type={type}
+            {...field}
+            onChange={(e) => {
+              handleChangeInput
+                ? handleChangeFc(e, handleChange)
+                : handleChange(e)
+            }}
```
Comment (phuc-create) [inline src/component/fieldtext/InputField.js]: I'm not sure the logic u are using here.

Comment (KNT2302) [reply src/component/fieldtext/InputField.js]: why are you not sure? tell me more detail please

File: src/component/fieldtext/InputField.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import { ErrorMessage, Field } from "formik"
+
+import React from "react"
+
+const InputField = ({ name, type, label, handleChangeInput }) => {
+  const handleChangeFc = (e, handleChange) => {
+    handleChangeInput(e, handleChange)
+  }
```
Comment (phuc-create) [inline src/component/fieldtext/InputField.js:8]: `const changeByType = (e, change) =>  type === "email" ?  change(e) : handleChangeInput(e, change)`

File: src/component/fieldtext/InputField.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        <div>
+          <label htmlFor={name}>{label}</label>
+          <input
+            type={type}
+            {...field}
+            onChange={(e) => {
+              handleChangeInput
+                ? handleChangeFc(e, handleChange)
+                : handleChange(e)
+            }}
```
Comment (phuc-create) [inline src/component/fieldtext/InputField.js]: `onChange={(e) => changeByType(e,handleChange) `

**Review by phuc-create — CHANGES_REQUESTED** [review]

File: src/component/fieldtext/InputField.js (hunk 2 above)
Comment (phuc-create) [reply src/component/fieldtext/InputField.js]: the logic u are using too complicated for other to understand

File: src/component/fieldtext/InputField.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    <Field name={name}>
+      {({ field, form: { handleChange } }) => (
+        <div>
+          <label htmlFor={name}>{label}</label>
+          <input
+            type={type}
+            {...field}
+            onChange={(e) => {
+              handleChangeType(e, handleChange)
+            }}
```
Comment (phuc-create) [inline src/component/fieldtext/InputField.js]: remove curly braces

**Review by phuc-create — CHANGES_REQUESTED** [review]

File: src/component/fieldtext/InputField.js (hunk 5 above)
Comment (KNT2302) [reply src/component/fieldtext/InputField.js]: removed

**Review by phuc-create — COMMENTED** [review]
image


check validation for email

File: src/component/form/FormInput.js [hunk 6]
```diff
@@ -22,7 +22,7 @@ const validationSchema = Yup.object({
     .max(100000, "Number is less than 100000")
     .required("Required"),
   emailField: Yup.string()
-    .matches(/^\w+@gmail.com$/i, "Email is invalid")
+    .matches(/^\w+([._-]?\w+)*@\w+([._-]?\w+)*(\.\w{2,})$/i, "Email is invalid")
     .required("Required"),
```
Comment (phuc-create) [inline src/component/form/FormInput.js:31]: If I have a task to validate the email like this,I will use this one :
`` /^[a-zA-Z0-9!#$%&''*+/=?^_`{}~@."\-\s]*$/ ``

**Review by phuc-create — APPROVED** [review]

File: src/component/form/FormInput.js (hunk 6 above)
Comment (phuc-create) [reply src/component/form/FormInput.js:31]: .matches(
    /^[a-zA-Z0-9!#$%&''*+/=?^_`{}~@."\-\s]*$/,
    'Please enter valid email address'
  )

**Pull request merged**